### ISSUE TITLE
kernel: export source-code-fixer.ts

### ISSUE THREAD

**Issue by azu** [textlint/textlint#735]
I'm creating https://github.com/textlint/editor.
It require `applyFixableMesage({ text, messsages })` like API.

Currently, @textlint/kernel proveides two APIs.

- `lintText(text: string, options: TextlintKernelOptions): Promise<TextlintResult>;`
- `fixText(text: string, options: TextlintKernelOptions): Promise<TextlintFixResult>;`

## Proposal

This proposal proposes a more primitive API.

- `applyMessages({ text: string, messages: TextlintMessage[] }): Promise<{ text: string })>`

It just exposes `sequentiallyApplyFixes` as public API. 

## Why

Currently, the external user like editor tool can not interop into fixing text.

Editor app like @textlint/editor do the following workflow

1. Lint text → get `TextlintResult`
2. Show `TextlintResult`'s `messages` as errors on the editor
3. Fix specific error messages

textlint does not expose "Fix specific error messages" API yet.
`fixText` API fix all text with all rules.

workaround: just text replace.

```
text.slice(0, message.fix.range[0]) + message.fix.text + text.slice(message.fix.range[1])
```

However, this workaround can not support multiple messages once.
This reason is that I've created `sequentiallyApplyFixes`.

### Another Approach

Split out `sequentiallyApplyFixes`  into a library.

## UseCase

- The user can fix text with a specific mesage
- The user can fix text with specific multiple mesages
   - In other words, The user can fix text with specific rule
- Allow to implement filtering for messages externally
   - @textlint/editor want to implement in-memory ignoring messages

**Comment by azu** [comment]
> Split out `sequentiallyApplyFixes` into a library.

I've selected this.

Because source-code-fixer has multiple functions.
So, I do not want to expose multiple API on the kernel.
